zcash_client_sqlite/src/lib.rs

```diff
@@ -579,8 +578,15 @@ impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
                 }
             }
 
+            // If any of the utxos spent in the transaction are ours, mark them as spent.
+            #[cfg(feature = "transparent-inputs")]
+            for txin in d_tx.tx.transparent_bundle().iter().flat_map(|b| b.vin.iter()) {
+                wallet::mark_transparent_utxo_spent(up, tx_ref, &txin.prevout)?;
```

Comment:
I confirmed that this uses `UPDATE` filtering on `prevout`, and therefore ignores spent TXOs that we aren't tracking.